llarp/router/outbound_session_maker.cpp

```diff
@@ -121,21 +121,28 @@ namespace llarp
   OutboundSessionMaker::ConnectToRandomRouters(int numDesired)
   {
     int remainingDesired = numDesired;
-
-    _nodedb->visit([&](const RouterContact &other) -> bool {
-      if(!_rcLookup->RemoteIsAllowed(other.pubkey))
-      {
-        return remainingDesired > 0;
-      }
-      if(randint() % 2 == 0
-         && !(_linkManager->HasSessionTo(other.pubkey)
-              || HavePendingSessionTo(other.pubkey)))
+    std::set< RouterID > exclude;
+    int itersLeft = 100;
```

Comment:
The purpose and consequences of `itersLeft` is pretty unclear, and `100` might be best defined as a constant

Reply:
it's to prevent infinite loops